cmd/dagger/terminal.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"sync"
+
+	tea "github.com/charmbracelet/bubbletea"
+)
+
+var terminalMu sync.Mutex
+
+func withTerminal(fn func(stdin io.Reader, stdout, stderr io.Writer) error) error {
+	// only allow one terminal session at a time
```

Comment:
If more than one come in at the same time does it just drop you into them consecutively? No problem with that, just confirming it's that vs. only one will be successful

Reply:
Yes, one after the other